### PR TITLE
[FLINK-25490][checkpoint] Complete the Chinese document regarding final checkpoint

### PR DESCRIPTION
**Pull request by gaoyunhaii**
## What is the purpose of the change

This PR completes the Chinese documentation regarding the final checkpoint. 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

File: docs/content.zh/docs/dev/datastream/fault-tolerance/checkpointing.md [hunk 1]
```diff
@@ -195,5 +195,44 @@ Flink 现在为没有迭代（iterations）的作业提供一致性的处理保
...
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+```
+
+在这种情况下，结束的任务不会参与 Checkpoint 的过程。在实现自定义的算子或者 UDF （用户自定义函数）时需要考虑这一点。
+
+为了支持部分任务结束后的 Checkpoint 操作，我们调整了 [任务的生命周期]({{<ref "docs/internals/task_lifecycle" >}}) 并且引入了
+{{< javadoc file="org/apache/flink/streaming/api/operators/StreamOperator.html#finish--" name="StreamOperator#finish" >}} 方法。
+在这一方法中，用户需要写出所有缓存的数据。在 finish 方法调用后的 Checkpoint 中，这一任务一定不能再有缓存的数据，因为在 `finish()` 后没有办法来输出这些数据。
+在大部分情况下，`finish()` 后这一任务的状态为空，唯一的例外是如果其中某些算子中包含外部系统事务的句柄（例如为了实现恰好一次语义），
+在这种情况下，在 `finish()` 后进行的 checkpoint 操作应该保留这些句柄，并且在结束 checkpoint （即任务退出前所等待的 checkpoint ）时提交。
```
Comment (zhuzhurk) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/checkpointing.md:216]: Maybe remove the spaces around the Chinese brackets？

File: docs/content.zh/docs/dev/datastream/fault-tolerance/checkpointing.md [hunk 2]
```diff
@@ -195,5 +195,44 @@ Flink 现在为没有迭代（iterations）的作业提供一致性的处理保
...
+Configuration config = new Configuration();
+config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, false);
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+```
+
+在这种情况下，结束的任务不会参与 Checkpoint 的过程。在实现自定义的算子或者 UDF （用户自定义函数）时需要考虑这一点。
+
+为了支持部分任务结束后的 Checkpoint 操作，我们调整了 [任务的生命周期]({{<ref "docs/internals/task_lifecycle" >}}) 并且引入了
+{{< javadoc file="org/apache/flink/streaming/api/operators/StreamOperator.html#finish--" name="StreamOperator#finish" >}} 方法。
+在这一方法中，用户需要写出所有缓存的数据。在 finish 方法调用后的 Checkpoint 中，这一任务一定不能再有缓存的数据，因为在 `finish()` 后没有办法来输出这些数据。
```
Comment (zhuzhurk) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/checkpointing.md:214]: maybe `缓存` -> `缓冲`?

**Review by zhuzhurk — COMMENTED** [review]
Thanks for adding this doc! 
It looks good to me except for 2 minor comments.

**Review by zhuzhurk — APPROVED** [review]
LGTM. +1 to merge.

**Comment by gaoyunhaii** [comment]
Thanks @zhuzhurk for the review! The failed test should be irrelevant, would merge...

**Pull request closed without merging**